pkg/api/file.go

```diff
@@ -198,6 +200,143 @@ func (s *server) fileUploadHandler(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// FileUploadInfo contains the data for a file to be uploaded
+type FileUploadInfo struct {
+	name        string // file name
+	size        int64  // file size
+	contentType string
+	toEncrypt   bool
+	reader      io.Reader
+}
+
+// GetFileHTTPInfo extracts data for a file to be uploaded from an HTTP request
+// this function was extracted from `fileUploadHandler` and should eventually replace its current code, along with `storeFile`
+func GetFileHTTPInfo(r *http.Request) (*FileUploadInfo, error) {
```

Comment:
This function is not used. There also is no reason to export it from the package.

From the comment is apparent that the intention was to use it in some next PR, however I see little point in adding a function right now. This PR would be cleaner without this one, especially since there are no tests for this function.

Also, the code seems to differ from the one about from which it was yanked. Instead of having 2 log lines before returning error (which did introduce a lot of duplication), there are now just opaque errors being returned. How will that look like when the function is used. Are the log lines useful.
There does not seem to be system for handling different error types. The original code returns HTTP 'BadRequest' and 'InternalServerError' responses. How are errors from this function to be translated to those different HTTP errors; or is it to be assumed that only a single HTTP error is returned.

P.S. Sorry for maybe a rant, I would remove this one for now, and save it for some later PR.

Reply:
> This function is not used. There also is no reason to export it from the package.
> 
> From the comment is apparent that the intention was to use it in some next PR, however I see little point in adding a function right now. This PR would be cleaner without this one, especially since there are no tests for this function.
> ...
> P.S. Sorry for maybe a rant, I would remove this one for now, and save it for some later PR.

i've removed this func per yours and @Eknir's [comment](https://github.com/ethersphere/bee/pull/447#discussion_r458681392), good points made all around.

> Also, the code seems to differ from the one about from which it was yanked. Instead of having 2 log lines before returning error (which did introduce a lot of duplication), there are now just opaque errors being returned. How will that look like when the function is used. Are the log lines useful.
There does not seem to be system for handling different error types. The original code returns HTTP 'BadRequest' and 'InternalServerError' responses. How are errors from this function to be translated to those different HTTP errors; or is it to be assumed that only a single HTTP error is returned.

i've thought about this too. we can definitely revisit these points in #462, but from my perspective:
- i don't really see a reason for having the 2 log lines, so this is why i removed it in the extracted code. this wasn't really a change in logic, just removing what's (in my opinion) unnecessary code. i am open to hearing arguments from the team as to why we need to call the logger twice.
- it is true that we are losing granularity in terms of error handling/visibility by thinning the API and extracting these functionalities. i can see why the HTTP response would not include the error, and just log it instead. this also affects tests.

maybe it's a good idea to re-think how errors are handled from the internal packages out to the API, and how visible they are once we are only giving back an HTTP response—whether it is a bad request, internal server error, or something else.

Reply:
I am not sure what is the exact changset but we have established these kind of guidelines about logging and errors:
1. Never log and return an error, do either one of those
2. The pair log.Debugf and log.Errorf is used together. Debugf usually contains more information and it is used for debugging, and is useful on clusters as well. log.Errorf is usually a more sensible message that will be visible for the user.

Reply:
i really don't think this logging practice is worth it.

having said that, i have added back the double log call to be consistent with the rest of the codebase.

Reply:
@mortelli @zbiljic reasons behind such logging is explained in https://github.com/ethersphere/bee/blob/master/CODING.md#logging.